contracts/controller/src/contract.rs

```diff
@@ -60,74 +61,78 @@ pub fn execute(
             };
             broker.accept_offer(deps, &offer)?;
             let send_msg = config.offer_denom.send(&info.sender, &offer.offer_amount);
-            let borrow_msg = vault_borrow_msg(&config.vault_address, offer.offer_amount)?;
-            let callback_msg = Controller(env.contract.address).call(
-                ExecuteMsg::UnstakeCallback {
-                    unbond_amount: config.ask_denom.coin(&amount),
+            let borrow_msg = vault_borrow_msg(
+                &config.vault_address,
+                offer.offer_amount,
+                Some(&CallbackType::Unstake {
                     offer,
-                },
-                vec![],
+                    unbond_amount: config.ask_denom.coin(&amount),
+                }),
             )?;
 
             Ok(Response::default()
                 .add_message(borrow_msg)
-                .add_message(send_msg)
-                .add_message(callback_msg))
+                .add_message(send_msg))
```

Comment:
Borrow funds and then send, or send then borrow funds, both should work in principle, up to you.